Support multi-project repositories

crywolf opened the pull request:
Some repositories host multiple projects that are related but have disjoint changelogs. This is typical for multi-crate workspaces in Rust. When configured as multi-project repo and using `project:` keyword in commit changelogs this feature allows to generate separate changelogs.

closes #14

crywolf committed:
Support multi-project repositories

Some repositories host multiple projects that are related but have disjoint changelogs. This is typical for multi-crate workspaces in Rust. When configured as multi-project repo and using `project:` keyword in commit changelogs this feature allows to generate separate changelogs.

changelog:
  section: features
  inherit: all

crywolf committed:
'project' arg can be empty when reading from stdin even in multi-prject repo (used in Github hook)

changelog: skip

Kixunil commented on wasm/src/lib.rs:
```diff
@@ -32,7 +32,7 @@ fn run(config: &str, git_callback: js_sys::Function) -> Result<(), Box<dyn std::
     let mut changelog = Changelog::new(&mut template, git);
 
-    changelog.generate()?;
+    changelog.generate(None)?;
```
Yeah, this is fine since the WASM module currently only supports `check` command.

crywolf committed:
Check modified files in multi-project setting

Massive improvement in the projects setting and rules for commits validation.

Projects in multi-project setting have directory assigned to each of them and files changed by commit are checked so that commit can change only files that belong to project specified in commit message.

    changelog:
        section: features
        inherit: title
        description: Projects in multi-project setting have directory assigned to each of them and files changed by commit are checked so that commit can change only files that belong to project specified in commit message.

crywolf committed:
pre-commit github hook

changelog: skip

crywolf committed:
Show commit template with hints

When user calls `git commit` a dynamic template based on YAML settings with hints about valid section names
(and prefiled project based on files changed in commit) is displayed to make the process of filling out necessary changelog section easier.

changelog:
    section: features
    inherit: all

crywolf committed:
Commit template (for githooks)

changelog: skip

crywolf committed:
Added Support section to README.md

changelog: skip

crywolf committed:
Update commit template

changelog: skip

crywolf committed:
Changelog hints in commit template are uncommented and nicely formatted

changelog: skip

crywolf commented:
Closing in favour of #21
